[CS2113T-M16-1] Task Scheduler

tqiren opened the pull request:
@joelczk @jichngan @lwxymere @Keith-JK

e0309556 commented:
@joelczk @jichngan @lwxymere @Keith-JK

joelczk committed:
Update README.md

joelczk committed:
Update README.md

lwxymere committed:
Fix broken formatting

lwxymere committed:
Update tasklist class diagram

lwxymere committed:
Merge pull request #148 from lwxymere/branch-DeveloperGuide

Fix broken links

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+. Locate and select the `build.gradle` file, then click `OK`
+. Click `Open as Project`
+. Click `OK` to use the default settings provided
+
```
Good to include the setting up process to lead the user 😄

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+* If there are tasks in the existing task list, it will call the `clearList()` method from the `TaskList` class to clear the
+existing taskList
+
+Given below is an example usage of `clear done` command:
```
Some suggestion over here: because you tried to explain this feature with 2 example, you may consider to split it with a title named "Example 1: usage of clear all" and "Example 2: usage of clear done" and bold it to make a clear separation? Because it can be confusing as both have similar steps 😃

lowjiayee replied:
Perhaps can highlight this sentence, to emphasize that this is the second example?

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+e.g. Typing `help` command and pressing kbd:[Enter] will list the commands present
+.. Some example commands you can try to get familiar with *ATAS*:
+* `help`: Lists the commands that *ATAS* supports.
+* `assignment n/Assignment One m/CS2113T d/01/01/20 1600 c/Important Assignment`: Adds an assignment called *Assignment
```
Maybe can include a section for adding the event / assignment ? In the POV of a developer, if the developer accidentally ignored this sentence, he / she will feel lost because there are no specific instruction in the implementation section until EDIT feature.

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+*Step 4* +
+The `execute()` method in the `EditCommand` class first gets an input from the user on the details of the edited task.
+
+[TIP]
```
Maybe the add feature / the parameters can be mentioned earlier / elaborate the adding function? 
Reason: to avoid potential confusion. Developer may be confused that add function is a sub function in Edit function

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+The following sequence diagram summarizes how repeat command operation works: +
+
+image::RepeatCommand_UML.png[Repeat Command Sequence Diagram]
```
Clear sequence diagram on explaining this features ! 👍 
Possible minor fix: The arrow (circle in red) is disjoint from the repeatCommand box.
![Screenshot (9)_LI](https://user-images.githubusercontent.com/42569361/77922851-2b1a5280-72d4-11ea-973c-94ae23918571.jpg)

lowjiayee replied:
Aside from what @jiajuinphoon has mentioned, the diagram looks good!

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+The following sequence diagram summarizes how delete command operation works: +
+
+image::delete.png[delete task]
```
Clear sequence diagram to show the features ! 👍 
Possible fix: Maybe can describe the if else condition with more words ? Because some may get confused about the [0] in the diagram 😺

GanapathySanathBalaji replied:
1) Maybe the condition label on the opt block can be improved to say that the list is empty instead of just mentioning a "0" for easier readability.

![image](https://user-images.githubusercontent.com/59698024/77986576-0660c200-734a-11ea-89e3-f48b9f245c3c.png)

GanapathySanathBalaji replied:
2) Is this correct? The change of control from the Parser object to the Delete Comand isn't shown for the second transition.

![Annotation 2020-03-31 122304](https://user-images.githubusercontent.com/59698024/77986767-871fbe00-734a-11ea-9751-7ad88ca2b6cf.jpg)

lowjiayee replied:
![image](https://user-images.githubusercontent.com/42570474/77986866-ca7a2c80-734a-11ea-8279-967b6abc256b.png)

1. I'm not sure if `opt` refers to the case with no task, or the case with tasks.
1. Would `alt` be a more accurate representation of the program flow, since only one of the two cases will be executed?

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+The following sequence diagram summarizes how the `ClearCommand` operation works: +
+
+image::clear.png[clear command]
```
CLear sequence diagram to describe this feature 👍 
Possible minor bug: The arrow seems to be placed at a wrong place ? it should be pointed to the top right corner. 
![Screenshot (10)_LI](https://user-images.githubusercontent.com/42569361/77989850-652a3980-7352-11ea-86fa-230160063a2e.jpg)

lowjiayee replied:
Perhaps can remove the whitespace to the right of this diagram? This should upscale the diagram such that it has similar font size as the DG.

GanapathySanathBalaji replied:
1. Is this correct? The control seems to change from TaskCommand to CommandResults even before the control reaches the function.

2. If the CommandResult class is used for the 3 different alternative branches, shouldn't it be constructed/deleted outside the alt block? If not the class needs to initialized/deleted in all three branches separately.

3. Shouldn't the self invocation arrows point to the top of the activation bar?

![Annotation 2020-03-31 123402](https://user-images.githubusercontent.com/59698024/77988313-9f91d780-734e-11ea-944d-b9543b7a4d05.jpg)

sitinadiah25 replied:
Should the arrow to of the self-invocation method be not pointing to the top right corner of the method? 
![image](https://user-images.githubusercontent.com/17756309/77988046-f6e37800-734d-11ea-965c-6e3e7b4226a3.png)
Here is the example given in our textbook: 
![image](https://user-images.githubusercontent.com/17756309/77988071-05ca2a80-734e-11ea-8a19-69b6ef5a3447.png)
Do check your other sequence diagrams for this error as well! 😅

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+=== Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user's schedule.
+
+image::TaskList Task class diagram.PNG[TaskList and Tasks]
```
Good on using composition and correct multiplicity in this diagram 👍

lowjiayee replied:
This diagram gives a very good understanding of your program! Also, all the arrows look correct to me!

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+The following sequence diagram summarizes how the `SearchCommand` and `SearchdCommand` works:
+
+image::search.png[Search operations]
```
Clear sequence diagram to show the features ! 👍

GanapathySanathBalaji replied:
1. Is the object's name written in the correct format? Shouldn't it be ":SearchCommand" instead of "SearchCommand"?
![Annotation 2020-03-31 122733](https://user-images.githubusercontent.com/59698024/77987096-758ae600-734b-11ea-94b8-d793275bd52a.jpg)

lowjiayee replied:
Perhaps can show the [alt] box when calling `resultList`, to represent the 2 different program flow paths?

sitinadiah25 replied:
The notes given on the sequence diagram helps the reader know what getSearchQueryX refers to and the assumptions that the diagram has. This makes it very clear and easy to understand to the reader 💯

jiajuinphoon commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+In particular, the diagram below shows the explicit execution flow that `CalendarCommand` takes.
+
+.Explicit execution flow of CalendarCommand
+image::addMonthlyCalendar.png[]
```
Good description on this feature by using a sub sequence diagram to show the features ! 👍

lowjiayee replied:
Perhaps if it is possible to prevent [opt] from being blocked by the underlying activation bar?

sitinadiah25 replied:
The use of a sub sequence diagram to zoom into the method is really useful for the developers. Good! 👍

jiajuinphoon reviewed (commented):
Overall, this is a clear and detail developer guide, despite of the missing of the adding features which can be confusing while reading it.  Clear sequence diagrams helped me to understand the features and the design consideration is easy to understand. However, the usage of several examples might confuse the developer / reader. Maybe can try to make the separation clearer to decrease the confusion ? Nonetheless a very good developer guide 👍 😄 ,

joelczk committed:
test DG

joelczk committed:
no message

joelczk committed:
no message

joelczk committed:
no message

joelczk committed:
Update DeveloperGuide.md

Ammended broken link

joelczk committed:
UG content page

joelczk committed:
Merge remote-tracking branch 'origin/Branch-UG' into Branch-UG

joelczk committed:
fix broken links

joelczk committed:
Update UserGuide.md

Wrong spelling

joelczk committed:
Merge pull request #149 from joelczk/Branch-UG

Add content table

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+The `Atas` component integrates all the aforementioned components to run the overall application logic. +
+The sequence diagram below shows how various components, broken down into the various classes, interact when the user enters a `help` command +
+
+image::atas help command sequence diagram v3.PNG[Component interactions for help command]
```
![image](https://user-images.githubusercontent.com/42570474/77986355-7753aa00-7349-11ea-9a8d-43f6cc9c41d1.png)

I'm not sure exactly when will `Storage` be called by `Atas`... Maybe can clarify if `Storage` is being called after every command, or before the program is terminated?

sitinadiah25 replied:
Good use of sequence diagram! Easy to read with the help of colours.

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+A summary of all the features available in *ATAS* can be found in <<Commands Summary>>. +
+Refer to <<Features>> for the detailed instruction of the various commands of *ATAS*.
+
+== Design
```
Would having an Introduction section before the Design section help readers understand the purpose of this document? Having the purpose, scope and target audience of the application may help readers understand what this application is made for and who it is targeting.

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
 
-## Design & Implementation
+Content
+- [Setting up](#1-setting-up)
```
Section 1 and section 3 both mentions setting up, could they be mentioned under the same section instead?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+*Step 4* +
+The `execute()` method will do 2 things:
+
+* If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error
```
> it will **initialize** a new ...

Perhaps you mean 'instantiate'?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+* If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error
+message indicating an empty task list
+* If there are tasks in the existing task list, the `DeleteCommand` class will call the `deleteTask()` method from the
+`TaskList` class to delete the task, based on the index. At the end of the execution, the `DeleteCommand` class will
+initialize a new `CommandResult` class that prints out the success message for task deletion.
```
> `DeleteCommand` class will **initialize** a new ...

Perhaps you mean 'instantiate`?

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+4. The `Ui` class is used to show the `CommandResult` message to the user. +
+5. The `Storage` object is used to save the new state of the application.
+
+== Setting Up
```
Maybe this section is not necessary? As this document is for developers and there is already another section in the beginning on how to set up the project on their computer. 
It also confused me when I looked at the table of content and saw two sections with the same heading 'Setting Up' so it may be a good idea to either remove this section and direct the developers to the UG on helping users set up the application or change the name of the section.

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+## 4. Implementation
+This section will detail how some noteworthy features are implemented.
+### 4.1. Delete Task Feature
+Current Implementation:  
```
Should there be a header for current implementation - ie. 4.1.1, as it divided into a subsection under search task feature, as section 4.2.1

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+image::delete.png[delete task]
+
+==== Design Considerations
+* Calling `remove()` method in `deleteTask()` command of `TaskList` method instead of calling `remove()` method within
+the `execute()` method of the `DeleteCommand` class
+** Pros: Easier implementation for other classes that requires the same use.
+** Cons: Increased coupling amongst classes, which makes it harder for testing.
+** Rationale: We decided to implement it in such a way because we feel that the effects of increased coupling in such a
+case is minimal and testing for related classes and methods are not affected much. Furthermore, such implementation also
+allows us to keep all the related commands to the list of tasks within a class which keeps our code cleaner.
```
Good to see design consideration for the readers to understand your decision!

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+=== Search task feature
+==== Current Implementation
+The search task feature is currently implemented in both `SearchCommand` class and `SearchdCommand` class. Both
```
Is it possible to change the name of `SearchdCommand` to something more distinct from `SearchCommand`? Personally, I still cannot understand their difference from the explanation...

lowjiayee replied:
Perhaps `SearchWithDateCommand`? Just a suggestion.

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+... Run the command `java -jar atas.jar`. You will be greeted with the welcome screen of *ATAS* in a few seconds.
+
+== Implementation
+This section will detail how some noteworthy features are implemented.
```
Would adding a section on the Add Task Feature be considered noteworthy too? As a developer, I would want to know how a task is added into the program, especially since the other features cannot be done without adding any tasks first.

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+=== Delete Task Feature
+
+Current Implementation: +
```
Good documentation of the implementation of the Delete Task feature, along with the sequence diagram to aid in the understanding. It goes through the step by step of what happens in the background along with the exceptions that may be thrown and what happens afterwards. 👍

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+The following sequence diagram summarizes how repeat command operation works:  
+
+![Repeat Command Sequence Diagram](images/RepeatCommand_UML.png)
```
In this UML diagram, the arrow representing return from method from CommandResult class to RepeatCommand class and arrow representing calling a method from RepeatCommand to CommandResult is not complete.

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+The `ClearCommand` inherits from the `Command` class and initializes the `clearParam` to check which clear function
+has to be executed
+
+Given below is an example usage of the `clear all` command:
```
Perhaps can highlight this sentence, to emphasize that this is the first example?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+*** If there are matching search queries in the search results, `execute` method will then create a new `CommandResults`
+class to print out the search results.
+
+Given below is an example usage of the `Searchd` command: +
```
Perhaps can highlight this sentence, to emphasize that this is the second example?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+through and `searchParam` to get the search query that the user inputs. It also has a `date` parameter to check the date
+that the users wants to search from
+
+Given below is an example usage of the `Search` command: +
```
Perhaps can highlight this sentence, to emphasize that this is the first example?

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+        Expanding number of `Calendar` rows. This will require the need to increase the number of `Calendar` columns to preserve the integrity of a traditional calendar view. However, this also is infeasible as our goal is to keep the calendar compact such that it does not need to fill the screen.
+
+## 4.7. Storage
+### 4.7.1. Implementation
```
Could the header font size of subsections be kept consistent?

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+* Creating 2 separate classes for `SearchCommand` and `SearchdCommand`
+** Rationale: +
+To create 2 separate commands so that users can filter their search query more easily.
+** Alternatives Considered: +
```
Maybe having a consistent format would be better? Having a full stop after every sentence 
e.g. 
![image](https://user-images.githubusercontent.com/17756309/77987799-5c833480-734d-11ea-9564-bd3f89af4255.png)
There should be a full stop at the end of the "Use a search..." and the Cons bullet point. 
This would make the document look more consistent.

GanapathySanathBalaji commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+The following sequence diagram summarises what happens when the `EditCommand` class is executed.
+
+image::EditCommand_SequenceDiagram.png[]
```
Shouldn't the self invocation arrows point to the start of the activation bar, instead of a point below it?

![Annotation 2020-03-31 124310](https://user-images.githubusercontent.com/59698024/77987804-5ee58e80-734d-11ea-9a9b-7957f34f1f64.jpg)

lowjiayee replied:
Is it possible to create border for the activation bars? Attached is my high-effort doodle to explain.

![image](https://user-images.githubusercontent.com/42570474/77988335-addff380-734e-11ea-94e7-61e59879b0ac.png)

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+Method `parseCommand()` from the `Parser` class will be called to parse the command provided. Through this method, we will be able to obtain
+ information to get integers `eventID`, `numOfPeriod` and the string `typeOfPeriod`. +
+- `eventID` identifies the task that the user wishes to repeat. +
+- `numOfPeriod` and `typeOfPeriod` (`d`, `w`, `m`, or `y`) specifies how often the user wants to repeat the event.
```
Perhaps can explain what are `d`, `w`, `m`, `y`? It took me some time to realise they mean day, week, month, year respectively.

lowjiayee replied:
Something like line370 looks better to me!

> `d` (days), `w` (weeks), `m` (months) or `y` (years)

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+The `CalendarCommand` class extends the `Command` class with methods to implement the necessary pre-processing to display an overview of tasks in the given date. The following sequence diagram outlines an example execution of `CalendarCommand` when it is called and the interaction it has with the relevant components.
+
+![Interaction of CalendarCommand and the various major components](images/calendar-diagram.png)
```
Method call back arrow missing for getTasksbyYearMonth()

harithadiv commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+In particular, the diagram below shows the explicit execution flow that `CalendarCommand` takes.
+
+![Explicit execution flow of CalendarCommand](images/addMonthlyCalendar.png)
```
Perhaps remove some methods to make the diagram clearer instead?

siuhian replied:
Maybe could split up the diagram to a few parts for easier understanding?

GanapathySanathBalaji reviewed (commented):
The overall DG was very good, well-formatted and easy to follow. Just a few minor issues with the UML diagrams. Could it be better if all the UML diagrams were standardized to follow a specific design? So that the report could look more cohesive.

harithadiv reviewed (commented):
Generally, a good DG with clear instructions and descriptions (:

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+[[calendar]]
+.Sample output of Calendar Command
+image::calendar2.png[]
```
Good use of screenshots for visualization 👍

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+Expanding number of `Calendar` rows. This will require the need to increase the number of `Calendar` columns to preserve the integrity of a traditional calendar view.
+However, this also is infeasible as our goal is to keep the calendar compact such that it does not need to fill the screen.
+
+=== Storage
```
Perhaps can consider adding a sequence diagram to explain the program flow?

sitinadiah25 commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+= A.T.A.S (Amazing Task and Assignment System) Developer Guide
+:site-section: UserGuide
+:toc:
```
Would numbering the table of content help the reader with the flow of the documentation?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+= A.T.A.S (Amazing Task and Assignment System) Developer Guide
+:site-section: UserGuide
+:toc:
+:toclevels: 4
+:toc-title: Contents
+:toc-placement: preamble
+:sectnums:
+:imagesDir: images
+:table-caption: Table
+:stylesDir: stylesheets
+:xrefstyle: full
+:experimental:
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:warning-caption: :warning:
+endif::[]
```
Perhaps can use numbering instead of bullet points in content page, for consistency with the actual headers in DG?

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+This section will give a high-level overview of how various components in *ATAS* function and interact with each other.
+
+=== Architecture
+image::overall architecture.PNG[overall architecture]
```
This diagram gives a very good understanding of your program!

lowjiayee commented on docs/DeveloperGuide.adoc:
```diff
@@ -0,0 +1,665 @@
+
+=== Storage Component
+
+image::storage.PNG[Storage Class Diagram]
```
This diagram gives a very good understanding of your program!

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
-# Developer Guide
+By: `Team M16-1` Since: `Jan 2020` License: `MIT`
 
-## Design & Implementation
```
The format of the page looks weird feels like there is some space wasted on both sides making it difficult to read. Maybe it would be good to change to a different theme?

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+        -   `exit`: Exits **ATAS**.
+
+## 2. Design
+This section will give a high-level overview of how various components in **ATAS** function and interact with each other.
```
It is not very clear what your application is trying to achieve. Maybe can add a short sentence summarising what your app does?

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+### 2.4. Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user’s schedule.
+
+![TaskList and Tasks](images/TaskList Task class diagram.PNG)
```
Your class diagram could maybe can to a solid lined box? Since the notes uses solid lines box

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+![TaskList and Tasks](images/TaskList Task class diagram.PNG)
+
+### 2.5. Storage Component
+![Storage Class Diagram](images/storage.PNG)
```
Since you are abstracting out the attributes, maybe can remove the attributes compartment? Will make the diagram more concise.

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+The `Atas` component integrates all the aforementioned components to run the overall application logic.  
+The sequence diagram below shows how various components, broken down into the various classes, interact when the user enters a `help` command  
+
+![Component interactions for help command](images/atas help command sequence diagram v3.PNG)
```
I think your UI activation bar should not have a break in between? Because it is waiting for the return result from "help".

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+5.  The `Storage` object is used to save the new state of the application.
+
+## 3. Setting Up
```
Maybe can be grouped with point 1 Setting Up?

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+**Step 2**  
+The user enters `search t\{TASK TYPE} n\{SEARCH QUERY}` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command provided.
+
+**Step 3**  
```
Maybe can break down to more steps? This step is very complicated and hard to digest.

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+**Step 2**  
+The user enters `searchd t\{TASK TYPE} n\{SEARCH QUERY} d\{DATE}` into the command line. Method `parseCommand()` from the `Parser` class will be called to parse the command provided.
+
+**Step 3**  
```
Similar to the comment above, maybe can make it more concise to make this step easier to understand?

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+The following sequence diagram summarizes how the `SearchCommand` and `SearchdCommand` works:
+
+![Search operations](images/search.png)
```
Maybe can add some lines to show how the CommandResult class is initialised?

siuhian commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,845 @@
+
+#### 4.6.2. Design Considerations
+
+-   Duplicating `Task` objects instead of keeping the `RepeatEvent` as a single entity like how it is stored in the `TaskList`.
```
I am not sure of this design works, but maybe you can use a class level attribute to differentiate between repeat events and task objects

Commit:
Add formating and missing links for UG due to conversion to md from adoc.

Commit:
Add formating and missing links for DG due to conversion to md from adoc.

Commit:
Update repeat command UML from comments received

Commit:
Fix broken links on content page

Commit:
Add back to top for UG and DG

Commit:
Add minor changes repeat event content for UG

tqiren committed:
Merge pull request #150 from e0309556/master

Fix UG and DG formatting issues from conversion to md from adoc

joelczk committed:
Fixed index bug for search commands of events and assignments

joelczk committed:
Merge pull request #151 from joelczk/master

Fixed index bug for search commands of events and assignments

joelczk committed:
Set theme jekyll-theme-cayman

joelczk committed:
no message

joelczk committed:
Merge remote-tracking branch 'origin/Branch-DG'

Keith-JK committed:
Refactor code to make methods shorter

joelczk committed:
Update search per bug report. add documentation page.

joelczk committed:
Update documentation guide.

joelczk committed:
Merge pull request #188 from joelczk/Branch-DG

update search, add documentation

joelczk committed:
Update Documentation.md

joelczk committed:
Update Documentation.md

joelczk committed:
update documentaion

joelczk committed:
Merge pull request #189 from joelczk/Branch-DG

update documentation

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

lwxymere committed:
Update regex to disallow the '|' and '/' characters, and fix issue #176

jichngan committed:
Fix bugs for editing Assignment and Event tasks.

jichngan committed:
Merge pull request #190 from jichngan/Branch-EditCommand-BugFixes

Fix bugs for editing Assignment and Event tasks.

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Fixed UG index bug. Fixed DG bugs. Closed issue #154.

joelczk committed:
Merge pull request #191 from joelczk/Branch-DG

Fixed UG index bug. Fixed DG bugs. Closed issue #154.

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

lwxymere committed:
Disallow user input fields to only contain spaces

lwxymere committed:
Remove all instances of multiple consecutive spaces from user input

lwxymere committed:
Merge branch 'master' into branch-V2BugFixes

lwxymere committed:
Merge pull request #192 from lwxymere/branch-V2BugFixes

Fix regex issues

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

lwxymere committed:
Add storage section

lwxymere committed:
Add note for special characters that cannot be used in user input

lwxymere committed:
Merge pull request #193 from lwxymere/branch-UserGuide

Update UG based on v2 bugfixes

lwxymere committed:
Fix uneven storage header size

lwxymere committed:
Remove spaces from image file names

lwxymere committed:
Change uml diagram dotted boxes to solid-line boxes

lwxymere committed:
Add instructions for adding tasks

jichngan committed:
Fix bug to display error when task index supplied is less than or equals to 0.

jichngan committed:
Fix bug which created a new event if RepeatedEvents are edited.

jichngan committed:
Fix regex for Assignment and Event command.

jichngan committed:
Fix Checkstyle Violations

jichngan committed:
Merge pull request #194 from jichngan/Branch-EditCommand-BugFixes

Branch edit command bug fixes

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Add appendix E for DG

joelczk committed:
Merge pull request #195 from joelczk/Branch-DG

Add appendix E for DG

Keith-JK committed:
Refactor CalendarCommand: input MMYY only + shorten methods

lwxymere committed:
Add DevOps section

lwxymere committed:
Merge branch 'master' into branch-DeveloperGuide

lwxymere committed:
Update section numbering

Keith-JK committed:
Improve CalendarCommand testing

lwxymere committed:
Merge pull request #196 from lwxymere/branch-DeveloperGuide

DG v2 bugfix updates

Keith-JK committed:
Update DG and UG on changes to CalendarCommand

Keith-JK committed:
Merge branch 'master' into branch-afterDG-refactor

Keith-JK committed:
update expected.txt

Keith-JK committed:
Merge branch 'branch-afterDG-refactor' of https://github.com/Keith-JK/tp into branch-afterDG-refactor

Keith-JK committed:
CalendarCommand refactor

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Keith-JK committed:
Corrected DG diagrams and add future enhancements

Keith-JK committed:
Update UG summary to reflect changes in CalendarCommand

Keith-JK committed:
Merge pull request #198 from Keith-JK/branch-DG-afterv2

Update UG DG

lwxymere committed:
Add PPP v1

lwxymere committed:
Merge branch 'master' into branch-ppp

lwxymere committed:
Merge pull request #199 from lwxymere/branch-ppp

Add PPP

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Commit:
Fixed relevant bugs reported during PE dry run

Commit:
Update UG and DG for Repeat Command.

Commit:
Add more information for Testing section in DG.

Commit:
Fix failed test case arising from changes in RepeatCommand implementation.

tqiren committed:
Merge pull request #200 from e0309556/branch-repeatingEvents

Update repeat command as per feedback from dry run PE

joelczk committed:
resolve issue #165

joelczk committed:
Fixed checkstyle

joelczk committed:
Fix IO test

joelczk committed:
Merge pull request #201 from joelczk/Branch-DG

resolve issue #165

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Keith-JK committed:
PPPv1 + More CalendarTest

Keith-JK committed:
Added Ui Test

Keith-JK committed:
Cleanup redundant lines of code and fixed test code style

Keith-JK committed:
Improve DoneCommand test code coverage

Keith-JK committed:
Improve test coverage for DeleteCommand

Keith-JK committed:
Rename methods in CalendarCommand

joelczk committed:
A few updates. refactor code and update ug,dg. add ppp

joelczk committed:
improved ppp

joelczk committed:
Merge pull request #202 from joelczk/Branch-DG

A few updates. refactor code and update ug,dg. add ppp

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

lwxymere committed:
Add event date limitations

lwxymere committed:
Update date command format section

jichngan committed:
Edit User Guide.

lwxymere committed:
Refactor some magic string literals in regex

lwxymere committed:
Merge pull request #203 from lwxymere/branch-V2BugFixes

Minor refactoring and UG update

jichngan committed:
Merge branch 'master' into Branch-DeveloperGuide

* master:
  Refactor some magic string literals in regex
  Update date command format section
  Add event date limitations

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Keith-JK committed:
Added Section A: Target user profile and Section D: non functional requirements

Keith-JK committed:
Edit pppv1

Keith-JK committed:
Fixed code style

Keith-JK committed:
Added pppv1. Added DG appendix. Added various Junit test

Added:

1. PPPv1
2. DG appendix A & appendix D
3. Ui test

Improved:

1. code coverage for DeleteCommand
2. code coverage for DoneCommand
3. code coverage for CalendarCommand

Keith-JK committed:
Added HelpTest

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Keith-JK committed:
Increase EditCommandTest code coverage

Keith-JK committed:
Removed * imports

Keith-JK committed:
Added HelpTest and increased EditCommandTest coverage

Added HelpTest and increased EditCommandTest coverage

jichngan committed:
Edit DG (Minor).

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

jichngan committed:
Merge branch 'master' into Branch-DeveloperGuide

* master:
  Removed * imports
  Increase EditCommandTest code coverage
  Added HelpTest
  Fixed code style
  Edit pppv1
  Added Section A: Target user profile and Section D: non functional requirements
  Rename methods in CalendarCommand
  Improve test coverage for DeleteCommand
  Improve DoneCommand test code coverage
  Cleanup redundant lines of code and fixed test code style
  Added Ui Test
  PPPv1 + More CalendarTest

jichngan committed:
Add new Sequence Diagrams for DG. Add new description to DG.

jichngan committed:
Add Appendix F for manual testing of program.

Keith-JK committed:
Correction to UML diagram

joelczk committed:
Merge pull request #206 from Keith-JK/branch-DG-correction

Correction to UML diagram

joelczk committed:
refactor tests for search

joelczk committed:
add javadocs

joelczk committed:
ADD AN MISSING PERIOD TO JAVADOCS

joelczk committed:
Merge pull request #207 from joelczk/Branch-DG

refactor tests for search

jichngan committed:
Merge branch 'master' into Branch-DeveloperGuide

* master:
  ADD AN MISSING PERIOD TO JAVADOCS
  add javadocs
  refactor tests for search
  Correction to UML diagram

jichngan committed:
Add Appendix F into DG.

jichngan committed:
Add Appendix F.

jichngan committed:
Add Bulk of Appendix F.

jichngan committed:
Update Search Command in UG.

jichngan committed:
Add Table Of Contents for DG.

jichngan committed:
Merge pull request #208 from jichngan/Branch-DeveloperGuide

Branch developer guide

lwxymere committed:
Update AboutUs link

lwxymere committed:
Update PPP links

joelczk committed:
Update DG. Change error message for search commands. updated dg diagram for search. Moved documentation to DG. Add missing symbols in PPP.

joelczk committed:
Update EXPECTED.txt

joelczk committed:
update test

joelczk committed:
Merge pull request #209 from joelczk/Branch-DG

Update DG. Change error message for search commands. updated dg diagr…

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

Commit:
Add individual Project Portfolio Page

Commit:
Add JUnit for RepeatCommand, fix some bugs for RepeatCommand and refactor some parts of code

Commit:
Merge branch 'master' into branch-PPP

tqiren committed:
Merge pull request #210 from e0309556/branch-PPP

Add PPP

Keith-JK committed:
Correct invalid date message for CalendarCommand

Keith-JK committed:
Fix corresponding test for CalendarCommand

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Merge pull request #211 from Keith-JK/branch-Calendarbug

Fix incorrect message for CalendarCommand

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Search command date. fix corresponding test

joelczk committed:
removed tests

joelczk committed:
update tests

joelczk committed:
Merge pull request #212 from joelczk/Branch-DG

Search message

jichngan committed:
Fix bug in editCommand which did not allow user to edit the time, comments of the task while keeping the same name.

jichngan committed:
Fix edit command prompt message.

jichngan committed:
Fix Bug which display success message when user changes task type of task to be edit. (i.e. Assignment task edited to be event task)
Error message to be displayed if the above is done.

jichngan committed:
Fix Checkstyle violations.

jichngan committed:
Fix EditCommand test error.

jichngan committed:
Merge pull request #213 from jichngan/Branch-EditCommand-BugFixes

Branch edit command bug fixes

Commit:
Add more details to UG.

tqiren committed:
Merge pull request #214 from e0309556/branch-UG

Add more details to UG.

Keith-JK committed:
Fix bug in DG link. Fixed possible bug in CalendarCommand

Keith-JK committed:
Fix max CalendarRows for month with 6 rows

Keith-JK committed:
Merge pull request #215 from Keith-JK/branch-finalpush

Fix bug in DG link. Fixed possible bug in CalendarCommand

joelczk committed:
Update UG,DG. Add images.

joelczk committed:
Merge pull request #216 from joelczk/Branch-DG

Update UG,DG. Add images.

joelczk committed:
Update joelczk.md

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Update UserGuide.md

joelczk committed:
Update UserGuide.md

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

jichngan committed:
Add Prompt to notify user that he is no longer in "editing" mode.

jichngan committed:
Update EditCommandTest to match the changes to editCommand.

jichngan committed:
Merge pull request #217 from jichngan/Branch-EditCommand-BugFixes

Branch edit command bug fixes

tqiren committed:
Update UserGuide.md

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

jichngan committed:
Edit EditCommand Implementation.

jichngan committed:
Add more test cases to Appendix F.

tqiren committed:
Update DeveloperGuide.md

jichngan committed:
Merge branch 'master' into Branch-DeveloperGuide

* master:
  Update DeveloperGuide.md

tqiren committed:
Update DeveloperGuide.md

jichngan committed:
Merge branch 'master' into Branch-DeveloperGuide

* master:
  Update DeveloperGuide.md

jichngan committed:
Merge pull request #218 from jichngan/Branch-DeveloperGuide

Branch developer guide

tqiren committed:
Update e0309556.md

tqiren committed:
Update e0309556.md

jichngan committed:
Add draft of PPP.

jichngan committed:
Add PPP first part

jichngan committed:
Merge branch 'master' into Branch-JichNgan-PPP

* master:
  Update e0309556.md

lwxymere committed:
Fix headers and numbering

lwxymere committed:
Merge master into branch

lwxymere committed:
Fix broken image link

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

jichngan committed:
Add PPP.

jichngan committed:
Update PPP.

jichngan committed:
Merge pull request #219 from jichngan/Branch-JichNgan-PPP

Branch jichngan ppp

lwxymere committed:
Fix minor errors and inconsistencies

lwxymere committed:
Merge branch 'master' into branch-DeveloperGuide

lwxymere committed:
Merge pull request #220 from lwxymere/branch-DeveloperGuide

Normalise DG formatting

jichngan committed:
Format UG.

jichngan committed:
Merge branch 'master' into Branch-UserGuide

* master:
  Fix minor errors and inconsistencies
  Fix broken image link
  Fix headers and numbering

jichngan committed:
Edit UG.

lwxymere committed:
Fix bug when manually editing save file

jichngan committed:
Edit UG.

lwxymere committed:
Merge pull request #221 from lwxymere/branch-V2BugFixes

Fix bug when manually editing save file

jichngan committed:
Edit UG. Final Version 1.0

joelczk committed:
Update DeveloperGuide.md

joelczk committed:
Update DeveloperGuide.md

joelczk committed:
Update DeveloperGuide.md

joelczk committed:
Update DeveloperGuide.md

lwxymere committed:
Update PPP

lwxymere committed:
Merge branch 'master' into branch-ppp

lwxymere committed:
Merge pull request #222 from lwxymere/branch-ppp

Update ppp

Keith-JK committed:
Minor change to PPP. Add test for EditCommand in Parser

Keith-JK committed:
Minor additions to DG/UG

Keith-JK committed:
Fix Javadoc

Keith-JK committed:
Merge pull request #223 from Keith-JK/branch-minorchange

Edit PPP. Additional basic Parser test. UG/DG update

lwxymere committed:
Test image size for github pages print to pdf

lwxymere committed:
Merge pull request #224 from lwxymere/branch-DeveloperGuide

Test image size for github pages print to pdf

jichngan committed:
Add ReadME for github.

jichngan committed:
Merge branch 'master' into Branch-UserGuide

* master:
  Test image size for github pages print to pdf
  Fix Javadoc
  Minor additions to DG/UG
  Minor change to PPP. Add test for EditCommand in Parser
  Update PPP
  Update DeveloperGuide.md
  Update DeveloperGuide.md
  Update DeveloperGuide.md
  Update DeveloperGuide.md
  Fix bug when manually editing save file

jichngan committed:
Add README for GitHub.

jichngan committed:
Merge pull request #225 from jichngan/Branch-UserGuide

Branch user guide

joelczk committed:
Update joelczk.md

tqiren committed:
Update UserGuide.md

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

joelczk committed:
Fixed UG calendar bug.

joelczk committed:
Merge pull request #226 from joelczk/Branch-UG

Fixed UG calendar bug.

joelczk committed:
Merge branch 'master' of https://github.com/AY1920S2-CS2113T-M16-1/tp

tqiren committed:
Update DeveloperGuide.md

tqiren committed:
Update e0309556.md

jichngan committed:
Edit PPP.

jichngan committed:
Merge pull request #227 from jichngan/Branch-JichNgan-PPP

Edit PPP.

tqiren committed:
Update DeveloperGuide.md

tqiren committed:
Update UserGuide.md

joelczk committed:
Update joelczk.md

joelczk committed:
Update PPP.

joelczk committed:
Merge pull request #228 from joelczk/master

update PPP

joelczk committed:
Update _config.yml

Keith-JK committed:
Change ppp name

Keith-JK committed:
Merge pull request #229 from Keith-JK/branch-minorchange

Change ppp name

Keith-JK committed:
Fix link in AboutUs.md

Keith-JK committed:
no message

Keith-JK committed:
no message

Keith-JK committed:
Merge pull request #230 from Keith-JK/branch-minorchange

Fix link in AboutUs.md

jichngan committed:
Add Contributions

jichngan committed:
Add User Guide Roles.

jichngan committed:
Merge pull request #231 from jichngan/Branch-UserGuide

Branch user guide
